Source/DafnyLanguageServer/Workspace/Notifications/VerificationDiagnosticsParams.cs

```diff
@@ -0,0 +1,664 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection.Metadata;
+using MediatR;
+using Microsoft.Boogie;
+using Microsoft.Dafny.LanguageServer.Language;
+using OmniSharp.Extensions.JsonRpc;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using Range = OmniSharp.Extensions.LanguageServer.Protocol.Models.Range;
+
+#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
+
+namespace Microsoft.Dafny.LanguageServer.Workspace.Notifications {
+  /// <summary>
+  /// DTO used to communicate the current compilation status to the LSP client.
+  /// </summary>
+  [Method(DafnyRequestNames.VerificationDiagnostics, Direction.ServerToClient)]
+  public class VerificationDiagnosticsParams : IRequest, IRequest<Unit> {
```

Comment:
This class is Public API, and after we agree on it, it should not be modified anymore, only completed because else future VSCode extensions would break.